From aa8a4c7507ee7ab6b93c8e8e9d50c67893bd5dc2 Mon Sep 17 00:00:00 2001
From: Brittany Wald <brittany.wald@joyent.com>
Date: Tue, 15 Aug 2017 21:13:47 +0000
Subject: [PATCH] MANTA-3371 triggers cannot be installed with any vnodes in
 read-only mode Reviewed by: Cody Peter Mello <cody.mello@joyent.com> Reviewed
 by: Richard Bradley <richard.bradley@joyent.com> Approved by: Cody Peter
 Mello <cody.mello@joyent.com>

---
 CHANGES.md   | 4 ++++
 lib/meta.js  | 4 +++-
 package.json | 2 +-
 3 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/CHANGES.md b/CHANGES.md
index 991eafb..9d2c250 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,5 +1,9 @@
 # Changelog
 
+## v.3.3.0
+
+* [MANTA-3371](https://devhub.joyent.com/jira/browse/MANTA-3371) triggers cannot be installed with any vnodes in read-only mode
+
 ## v3.2.0
 
 * [MORAY-104](http://smartos.org/bugview/MORAY-104) disallow filtering on non-indexed fields except interactively
diff --git a/lib/meta.js b/lib/meta.js
index acae4bb..48b35d9 100644
--- a/lib/meta.js
+++ b/lib/meta.js
@@ -143,10 +143,12 @@ function sql(rpcctx, statement, values, options) {
     assert.ok(Array.isArray(values));
     assert.object(options, 'options');
     assert.optionalNumber(options.timeout, 'options.timeout');
+    assert.optionalBool(options.readOnlyOverride, 'options.readOnlyOverride');
 
     opts = {
         req_id: options.req_id || libuuid.create(),
-        timeout: options.timeout
+        timeout: options.timeout,
+        readOnlyOverride: options.readOnlyOverride || false
     };
     log = rpc.childLogger(rpcctx, opts);
     res = new events.EventEmitter();
diff --git a/package.json b/package.json
index 5887cdc..9350762 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
     "name": "moray",
     "description": "Moray client library",
-    "version": "3.2.1",
+    "version": "3.3.0",
     "author": "Joyent (joyent.com)",
     "keywords": [ "moray" ],
     "main": "./lib/index.js",
-- 
2.21.0

